src/pip/_internal/cache.py

```diff
@@ -228,7 +225,9 @@ def get(
             return link
 
         canonical_package_name = canonicalize_name(package_name)
-        for wheel_name in self._get_candidates(link, canonical_package_name):
+        for wheel_name, wheel_path in self._get_candidates(
+            link, canonical_package_name
+        ):
```

Comment:
nitpick: a simple ·`wheel_name = os.path.basename(wheel_path)` would remove the need to return the filename and its path in `_get_candidates`

Reply:
@xavfernandez Good points. I handled this by deferring `os.path.join` to the last moment so it becomes more efficient and hopefully more readable too.